File: src/main/java/izone/batch/BatchWordCount.java
package izone.batch;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.DataSet;
import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.AggregateOperator;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.util.Collector;

/**
 * Created by dell on 2019/9/15.
 */
public class BatchWordCount {
    public static void main(String[] args)  throws Exception {
        String input = "C:\\Users\\dell\\Desktop\\data.txt";
        String ouput = "F:\\result";

        //获取运行环境
        ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
        //获取文件中的内容
        DataSource<String> text = env.readTextFile(input);

        DataSet<Tuple2<String, Integer>> counts = text.flatMap(new Tokenizer()).groupBy(0).sum(1);

        counts.writeAsText(ouput).setParallelism(1);

        env.execute("batch word count");
    }

    public static class Tokenizer implements FlatMapFunction<String,Tuple2<String,Integer>> {
        public void flatMap(String value, Collector<Tuple2<String, Integer>> out) throws Exception {
            String[] tokens = value.toLowerCase().split("\\W+");
            for (String token: tokens) {
                if(token.length()>0){
                    out.collect(new Tuple2<String, Integer>(token,1));
                }
            }
        }
    }
}

File: src/main/scala/izone/batch/BatchWordCount.scala
package izone.batch

import org.apache.flink.api.scala.{DataSet, ExecutionEnvironment}

/**
  * Created by dell on 2019/9/15.
  */
object BatchWordCount {
  def main(args: Array[String]): Unit = {
    val env: ExecutionEnvironment = ExecutionEnvironment.getExecutionEnvironment

    val input: String = "C:\\Users\\dell\\Desktop\\data.txt"
    val output: String = "F:\\result.txt"

    val text: DataSet[String] = env.readTextFile(input)

    import org.apache.flink.api.scala._

    val sum: DataSet[(String, Int)] = text.flatMap(_.toLowerCase.split(" "))
      .filter(_.nonEmpty)
      .map((_, 1))
      .groupBy(0)
      .sum(1)

    sum.writeAsText(output).setParallelism(1)

    env.execute("batch word count")
  }
}
